src/packet/signature/config.rs

```diff
@@ -299,13 +299,18 @@ impl SignatureConfig {
         }
     }
 
-    /// Returns an iterator over all subpackets of this signature.
+    /// Returns an iterator over the hashed subpackets of this signature.
     pub fn subpackets(&self) -> impl Iterator<Item = &Subpacket> {
         self.hashed_subpackets
             .iter()
             .chain(self.unhashed_subpackets.iter())
```

Comment:
Wait... are you still returning unhashed packets here? :thinking: 

(The idea behind the change looks good, though).

Reply:
Ouch, thanks for checking. Not sure how I managed to commit this confused version :sweat_smile: 
I pushed an amended fix.